src/esbmc/bmc.h

```diff
@@ -38,6 +43,9 @@ class bmct
 protected:
   const contextt &context;
   namespacet ns;
+  // for multi-kind/incr
+  std::set<std::pair<std::string, std::string>> &to_remove_claims;
+  std::set<std::pair<std::string, std::string>> dump = {};
```

Comment:
I don't understand what purpose these member serve exactly. Could you please explain in a comment what they contain and where it's used?

Also, since this `std::pair<std::string, std::string>` is used with a particular meaning at multiple locations (first: claim_msg, second: claim_loc), I think it makes sense to introduce a new type that has these two strings as members, in order to differentiate those objects from arbitrary pairs of strings (such as those held by, for instance, a `std::map<std::string, std::string>`).